### DISCUSSION TITLE
Analyzing Component usage

### DISCUSSION THREAD

**Discussion by gfellerph** [swisspost/design-system#2231] [Ideas]
It would be awesome to track usage of web components throughout our applications. In the best case, without the need to change the implementation. This discussion is intended to clarify the boundaries and possibilities and other aspects around collecting such data.

**What data should be collected?**
- Anonymous usage data for web components
  - What components are being used?
  - What version do they have?
  - What settings are used?
  - Generic site data like user agent, device type, current URL

**How should this data be collected?**
There are two possible approaches.

1. Use the google tag manager and send custom events whenever a component loads, aggregate through the company wide analytics suite
  1.1 If google analytics is not implemented by the project, no data can be collected
  1.2 Projects have to potentially deal with "noise" events on their side
  1.3 Less legal concerns because consent is already given
2. Use a custom implementation
  2.1 How does the legal side of things look here? Are we allowed to collect this kind of data? Does the user have to agree to this?
  2.2 [Beacons](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon) could be a nice way to submit the data

**How do we intend to use this data?**
- See if a component or some variant of a component is not used at all to identify optimization potential
- Measure the update-willingness of projects
- Estimate impact of breaking changes

**Comment by gfellerph** [comment]
Discussion on 09.11.2023:
- We could implement our own GTM and send events there (should not trigger a separate consent banner). Could be an issue if GTM is not used by the project
- We should give the possibility to turn analytics off on a project level
- Corporate analytics is all in one place (dev/int/prod of all services), lot of noise anyways
- We need a clear strategy for usage of the data before implementing analytics
- We should make sure that data is not skewed by big projects
- Usage could be statically analyzed on bitbucket for internal projects (would produce different data)
- Gather feedback through the docs on deprecations or new features before they are shipped

**Comment by imagoiq** [comment]
Can we use artifactory as well to get some stats?

**Reply by gfellerph** [reply]
Only number of downloads per version as far as I could see.

**Comment by imagoiq** [comment]
I didn't check how it's working, but for inspiration at least It could be interesting to see what you can do with: https://luroapp.com/

**Reply by gfellerph** [reply]
The product itself looks amazing, although I believe setup would take some time. Would be interesting to do a little testing with that and see if it really can log in to some of our pages behind proxies etc. Otherwise it will have really limited use. The licensing fees could be quite high for custom solutions like ours with 100s of products with 10'000s of pages to crawl.

**Comment by gfellerph** [comment]
Discussion with analytics team lead:
Projects would need user data from components as well (tabs clicked, tooltips shown, etc.). Therefore it would be beneficial for us and projects to implement centralised analytics for every component through GTM and then create custom Design System dashboards.
